src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2086,7 +2086,22 @@ def derive_scrypt(self, key_material, salt, length, n, r, p):
             key_material, len(key_material), salt, len(salt), n, r, p,
             scrypt._MEM_LIMIT, buf, length
         )
-        self.openssl_assert(res == 1)
+        if res != 1:
+            errors = self._consume_errors()
+            if not self._lib.CRYPTOGRAPHY_OPENSSL_LESS_THAN_111:
+                # This error is only added to the stack in 1.1.1+
+                self.openssl_assert(
+                    errors[0]._lib_reason_match(
+                        self._lib.ERR_LIB_EVP,
+                        self._lib.ERR_R_MALLOC_FAILURE
+                    )
+                )
+
+            min_memory = 128 * n * r // (1024**2)
```

Comment:
Please add a comment with a cite to this formula; hopefully there's a clear reference?